### ISSUE TITLE
Check if ID already exists, and if so, modify it

### ISSUE THREAD

**Issue by Lela810** [Lela810/dashy#5]
Check if ID already exists, and if so, modify it

https://github.com/Lela810/dashy/blob/1814438a0b1315b458c39fd08a13d52c2401170f/src/utils/MiscHelpers.js#L42

```javascript

  return string.replace(reg, (match) => (map[match]));
};

<<<<<<< HEAD
/* Based on section title, item name and index, return a string value for ID */
const makeItemId = (sectionStr, itemStr, index) => {
  const charSum = sectionStr.split('').map((a) => a.charCodeAt(0)).reduce((x, y) => x + y);
  const itemTitleStr = itemStr.replace(/\s+/g, '-').replace(/[^a-zA-Z ]/g, '').toLowerCase();
  return `${index}_${charSum}_${itemTitleStr}`;
};

/* Given an array of sections, apply a unique ID to each item, and return modified array */
export const applyItemId = (inputSections) => {
  const sections = inputSections || [];
  sections.forEach((sec, secIdx) => {
    if (sec.items) {
      sec.items.forEach((item, itemIdx) => {
        sections[secIdx].items[itemIdx].id = makeItemId(sec.name, item.title, itemIdx);
        // TODO: Check if ID already exists, and if so, modify it
      });
    }
  });
  return sections;
=======
/* Given a timestamp, returns formatted date, in local format */
export const timestampToDate = (timestamp) => {
  const localFormat = navigator.language;

```

c5a46c8b78d6408545afab8125290d9f6bde2cd8

**Comment by Lela810** [comment]
If you're enjoying Dashy, consider dropping us a ⭐<br>_<sub>🤖 I'm a bot, and this message was automated</sub>_

**Comment by Lela810** [comment]
Closed in 92402df5629cc37f33054c50561d6d763d5c0c8a
